### DISCUSSION TITLE
Create Custom Popup

### DISCUSSION THREAD

**Discussion by G-Michel** [nhn/tui.calendar#1338] [Q&A] [answered]
Hello, 

The possibility to create our own popups seems to be removed since 2.XX versions, 

Before an event, "beforeCreateSchedule" was available and was used to create our own event popup.

it is not present in 2.XX versions.

or maybe there is a workaround or an another tip to address this issue ? 

Thanks :)

**Comment by adhrinae** [comment, marked as answer]
There was a major change in API naming.

https://github.com/nhn/tui.calendar/blob/main/docs/en/guide/migration-guide-v2.md#change-term-from-schedule-to-event

So you can use `beforeCreateEvent` instead of `beforeCreateSchedule`.

https://github.com/nhn/tui.calendar/blob/main/docs/en/apis/calendar.md#beforecreateevent
